### DISCUSSION TITLE
Distributing big data in thread mesh networks?

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [openthread/openthread#8470] [Q&A] [answered]
Our company is just evaluating the usability of thread concerning sending big amounts of data over a thread mesh. Big meaning OTA updates for embedded devices that range from ~2MB to maybe 6MB. Can this be done easily via thread? Also our goal is to distribute JSON data from cloud to end device via mesh. I see that thread supports TCP which would be a must in our case. Is thread the right pick for this use case?

**Comment by jwhui** [comment, marked as answer]
The latest Thread 1.3 Specification does have provisions for bulk transfer, including TCP and delay-aware queueing. Note, however, that Thread was designed for low data-rate applications. If transfer latency is not a concern, then Thread may be a good fit for your application.

**Comment by MartyMcFlyInTheSky** [comment]
Thanks! Well the updates are not necessarily time constrained like a video stream for example. But how fast would the distribution of say a 2MB file be as a ballpark figure? Is there some data on transfer latency vs. chunk size?

**Reply by jwhui** [reply]
@samkumar's [NSDI'20 paper](https://www.usenix.org/conference/nsdi20/presentation/kumar) has data showing goodput of 75 kb/s over single hop and 20 kb/s over multiple hops. Take a look at the paper for more details.
